tools/test_modules/m21500.pm

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env perl
+
+##
+## Author......: See docs/credits.txt
+## License.....: MIT
+##
+
+use strict;
+use warnings;
+
+use MIME::Base64 qw (encode_base64 decode_base64);
+use Crypt::PBKDF2;
+use Digest::SHA qw (sha512);
+use Encode;
+
+sub module_constraints { [[0, 256], [8, 16], [-1, -1], [-1, -1], [-1, -1]] }
```

Comment:
From what I can see with JtR mode is that the length of the hash is always the same. But here you use dynamic salt length 8 - 16. How does that match together?

Reply:
Because in the details is written:

> The hash is computed by first generating a salt that consists of the lowercase username. If the salt is less than 8 bytes long, it is appended with bytes from the string "1244352345234" until it is 8 bytes. For example, the salt for username "ADMIN" would become "admin124", while the salt for "Bo" would become "bo124435".

I don't see any reference to fixed len of salt. The "username" length is dynamic from my point of view

Reply:
ok

Reply:
jtr handle also the padding of the salt/username but I wanted to hear your opinion first :)